spring-cloud-previews/google-cloud-language-spring-starter/src/test/java/com/google/cloud/language/v1/spring/LanguageAutoConfigurationTests.java

```diff
@@ -104,6 +105,25 @@ void testCredentials_fromTopLevelIfNoServiceProperties() {
             });
   }
 
+  @Test
+  void testShouldGetTransportChannelProviderFromBeanWithQualifierName() throws IOException {
+    this.contextRunner
+        .withBean(
+            "anotherTransportChannelProvider",
+            TransportChannelProvider.class,
+            () -> mockTransportChannelProvider)
+        .run(
+            ctx -> {
+              LanguageServiceClient client = ctx.getBean(LanguageServiceClient.class);
+              TransportChannelProvider transportChannelProviderBean =
+                  (TransportChannelProvider) ctx.getBean(TRANSPORT_CHANNEL_PROVIDER_QUALIFIER_NAME);
```

Comment:
I guess you can also check that there are two `TransportChannelProvider` beans in the context, just to verify that the setup is correct.

Reply:
Good idea. updated.